services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,869 @@
+        flowPathInfoAfterUpdating.protectedPath
+
+        currentLastUpdate < northbound.getFlow(flow.id).lastUpdated
+
+        and: "Rules for main and protected paths are created"
+        Wrappers.wait(WAIT_OFFSET) { flowHelper.verifyRulesOnProtectedFlow(flow.id) }
+
+        when: "Update flow: disable protected path(allocateProtectedPath=false)"
+        def protectedFlowPath = northbound.getFlowPath(flow.id).protectedPath.forwardPath
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = false })
+
+        then: "Protected path is disabled"
+        !northbound.getFlowPath(flow.id).protectedPath
+
+        and: "Rules for protected path are deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            protectedFlowPath.each { sw ->
+                def rules = northbound.getSwitchRules(sw.switchId).flowEntries.findAll {
+                    !Cookie.isDefaultRule(it.cookie)
+                }
+
+                assert flowHelper.findForwardPathRules(rules, sw).size() == 0
+                assert flowHelper.findReversePathRules(rules, sw).size() == 0
+            }
+        }
+
+        and: "Cleanup: delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "Unable to create a single switch flow with protected path"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+
+        when: "Create single switch flow"
+        def flow = flowHelper.singleSwitchFlow(sw)
+        flow.allocateProtectedPath = true
+        flowHelper.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Couldn't setup protected path for one-switch flow"
+    }
+
+    def "Unable to update a single switch flow to enable protected path"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+
+        and: "A flow without protected path"
+        def flow = flowHelper.singleSwitchFlow(sw)
+        flowHelper.addFlow(flow)
+
+        when: "Update flow: enable protected path"
+        northbound.updateFlow(flow.id, flow.tap { it.allocateProtectedPath = true })
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not update flow: Couldn't setup protected path for one-switch flow"
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    @Unroll
+    def "System is able to switch #flowDescription flow to protected path"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def allIsls = northbound.getAllLinks()
+        def (Switch srcSwitch, Switch dstSwitch) = getNotNeighboringSwitchPair(2)
```

Comment:
should be `getNotNeighboringSwitchPair(3)` because on L198-199 you expect that new protected path will be present